userspace/libsinsp/fdinfo.cpp

```diff
@@ -425,3 +475,23 @@ void sinsp_fdtable::lookup_device(sinsp_fdinfo* fdi, uint64_t fd)
 	}
 #endif // _WIN32
 }
+
+sinsp_fdinfo* sinsp_fdtable::find(int64_t fd)
+{
+	return find_ref(fd).get();
+}
+
+sinsp_fdinfo* sinsp_fdtable::add(int64_t fd, std::unique_ptr<sinsp_fdinfo> fdinfo)
+{
+	return add_ref(fd, std::move(fdinfo)).get();
```

Comment:
Note -- this is done in order to inline the shared_ptr access and avoid reference counting in the hot path for all code not using the state API. This helps us ensuring that all existing use cases have no performance impact. This has been also validated with few benchmarks on my side.